### PR TITLE
Update installed compiler dependencies

### PR DESCRIPTION
**Pull request by andjo403**
As llvm have updated the minimum toolchain
https://reviews.llvm.org/D66188

### CODE REVIEW

**Comment by nikic** [comment]
The bump only happened for LLVM 10 (while we're on LLVM 9), so technically we still support older compiler versions at this point in time.

**Comment by andjo403** [comment]
but to get it to work you need to set the allow-old-toolchain in the config.toml for llvm to receive LLVM_TEMPORARILY_ALLOW_OLD_TOOLCHAIN shall we add that to the description then?

**Comment by JohnCSimon** [comment]
Ping from triage - 
Hello, this PR has sat idle for eight days, what else needs to happen to resolve this?
@nikic @andjo403 @nikomatsakis 

Thank you.

**Comment by andjo403** [comment]
maybe I shall point to the https://reviews.llvm.org/D57264 that is in the llvm version used and due to that commit there is an compile error if the LLVM_TEMPORARILY_ALLOW_OLD_TOOLCHAIN option is not set and as that is not described in the readme I think update the compiler version is a good solution when llvm 10 also removes that possibility anyway.

**Comment by nikomatsakis** [comment]
I think it makes sense to list the more modern dependencies as the default, and perhaps add a "subbullet" or something saying "you can use the older versions if you set LLVM_TEMPORARILY_ALLOW_OLD_TOOLCHAIN". (IIUC, that env variable is needed to use the older versions?)

**Comment by andjo403** [comment]
added a description in new chapter "Building rustc with older host toolchains"

File: README.md [hunk 1]
```diff
@@ -148,6 +148,17 @@ by manually calling the appropriate vcvars file before running the bootstrap.
 > python x.py build
 ```
 
+### Building rustc with older host toolchains
+between the below versions of the host toolchains the rustc build fails unless
```
Comment (nikomatsakis) [inline README.md]:
```suggestion
It is still possible to build Rust with the older toolchain versions listed below, but only if the
```

File: README.md [hunk 2]
```diff
@@ -148,6 +148,17 @@ by manually calling the appropriate vcvars file before running the bootstrap.
...
+LLVM_TEMPORARILY_ALLOW_OLD_TOOLCHAIN option is set to true in the config.toml file
+
+* Clang 3.1 - 3.5
+* Apple Clang 3.1 - 6.0
+* GCC 4.8 - 5.1
+* Visual Studio 2015 (Update 3) - Visual Studio 2017
+
+with older versions then the above it is not possible to build rustc.
```
Comment (nikomatsakis) [inline README.md]:
```suggestion
Toolchain versions older than what is listed above cannot be used to build rustc.
```

**Review by nikomatsakis — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (andjo403) [reply README.md]: used between due to the upper version do not need the option to be set but was not that happy with that wording
shall I try to find the upper version the is not working and add them in the list below?

**Comment by JohnCSimon** [comment]
Ping from triage -
Hi! @andjo403 , can you please address the change request from @nikomatsakis?

Thank you.

**Comment by andjo403** [comment]
update the text according to the comments but also removed the range of compilers that was listed before due to the higher version listed did not need the option to be set

**Comment by nikomatsakis** [comment]
@bors r+ rollup

Thanks @andjo403

**Pull request merged**